tree/tree/src/TEntryList.cxx

```diff
@@ -512,6 +512,61 @@ void TEntryList::Add(const TEntryList *elist)
 
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add a sub entry list to the current list.
+/// \param[in] elist an entry list that should be added as a sub list of this list.
+///
+/// This function is specifically targeted at situations where there is a global
+/// TEntryList that should hold a bunch of sub TEntryList objects. For example,
+/// if one wants to create a one to one mapping between the sub entry lists and
+/// the trees in the files that make a TChain. Note that in such cases this
+/// configuration of the entry list should be used in pair with the option \p "sync"
+/// of the function TChain::SetEntryList
+///
+/// ~~~{.cpp}
+/// // Create a TChain with two files. Each contains a tree with 20 entries
+/// TChain chain{"entries"};
+/// chain.Add("file_20entries_1.root");
+/// chain.Add("file_20entries_2.root");
+///
+/// // Create a global, empty TEntryList. 
+/// TEntryList elists;
+/// // Create two entry lists. Each one will be referring to a different tree in the chain
+/// TEntryList elist1{"","","entries","file_20entries_1.root"};
+/// TEntryList elist2{"","","entries","file_20entries_2.root"};
+///
+/// // Select the first ten entries from the first tree and all entries from the second
+/// for(auto entry = 0; entry < 10; entry++){
+///     elist1.Enter(entry);
+/// }
+/// for(auto entry = 0; entry < 20; entry++){
+///     elist2.Enter(entry);
+/// }
+///
+/// // Add sub entry lists to the global list
+/// elists.AddSubList(elist1);
+/// elists.AddSubList(elist2);
+///
+/// // Set the entry list in the chain. Note the usage of option "sync"
+/// chain.SetEntryList(&elists, "sync"); 
+/// ~~~
+
+void TEntryList::AddSubList(const TEntryList &elist){
+
+   // This was the only way I could find to avoid errors like
+   // Error in <TList::Delete>: A list is accessing an object (0x7ffc67ad3540) already deleted (list name = TList)
+   // at runtime. Preferrably we wouldn't need to copy the argument elist into
+   // a new TEntryList inside this function
+   auto elistcopy = new TEntryList{elist};
+
+   fN += elistcopy->fN;
+
+   if (!fLists){
+      fLists = new TList();
+   }
+   fLists->Add(elistcopy);
+}
```

Comment:
this logic is much simpler that what is done in `Add`. One thing I see it's missing is handling the case in which `elist` has sublists (totally ok on my side, maybe it warrants a mention in the method description).

@pcanal do we need to take care of something else that `Add` takes care of?

Reply:
With the current signature there is nothing more to do.  The complexity in the regular case is the "or merge with existing" part.  Since here, by contract, it is just an add, we are done.  (One would need to reintroduce that complexity, if this routine was 'AddAt' to merge with an existing entry list as the same index.